public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,303 @@
+    ((filterInfo.placeholder2 !== undefined && config.filterExpression2 !== '') ||
+      filterInfo.placeholder2 === undefined) &&
+    !fieldNameInvalid &&
+    !filterTypeInvalid &&
+    !filterValid;
+
+  return (
+    <div className="gf-form-inline">
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-4">Field</div>
+        <Select
+          className="width-24"
+          placeholder="Field Name"
+          options={fieldNameOptions}
+          value={config.fieldName}
+          invalid={fieldNameInvalid}
+          onChange={value => {
+            onConfigChange({
+              ...config,
+              fieldName: value?.value ?? null,
+            });
+          }}
+          isClearable
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Match</div>
+        <Select
+          invalid={filterTypeInvalid}
+          className="width-8"
+          placeholder="Select test"
+          options={filterTypeOptions}
+          value={config.filterType}
+          onChange={value => {
+            onConfigChange({ ...config, filterType: (value.value as ValueFilterID) ?? ValueFilterID.regex });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form--grow gf-form-spacing ">
+        {filterInfo.placeholder && (
+          <Input
+            className="flex-grow-1"
+            invalid={filterInstance.expression1Invalid ?? filterExpressionInvalid}
+            defaultValue={config.filterExpression || undefined}
+            placeholder={filterInfo.placeholder}
+            onBlur={event => {
+              onConfigChange({ ...config, filterExpression: event.currentTarget.value });
+            }}
+          />
+        )}
+      </div>
+      {filterInfo.placeholder2 && (
+        <div className="gf-form gf-form-spacing gf-form--grow">
+          <Input
+            className="flex-grow-1"
+            invalid={filterInstance.expression2Invalid}
+            defaultValue={config.filterExpression2 || undefined}
+            placeholder={filterInfo.placeholder2}
+            onBlur={event => {
+              onConfigChange({ ...config, filterExpression2: event.currentTarget.value });
+            }}
+          />
+        </div>
+      )}
+      <div className="gf-form">
+        <Button icon="times" onClick={onDelete} style={{ height: '100%' }} size="sm" variant="secondary" />
+      </div>
+    </div>
+  );
+};
+
+export const FilterByValueTransformerEditor: React.FC<TransformerUIProps<FilterByValueTransformerOptions>> = ({
+  input,
+  options,
+  onChange,
+}) => {
+  const styles = getEditorStyles();
+  const fieldsInfo = useMemo(() => getAllFieldInfoFromDataFrames(input), [input]);
+  const fieldNameOptions = useMemo(
+    () =>
+      fieldsInfo
+        .filter((item: Record<string, any>) => item.type !== FieldType.time)
```

Comment:
@Totalus just so I understans. Why do we want to exclude the time field?

Reply:
I commented on that before:

>There is one thing I would improve. As it is right now, it is possible to apply a few filter types on a time field, but the filtering will be confusing for the user. The time is displayed in an text format (ISO or other), but the filtering is applied on the numeric (epoch) value.
>
> I see two possibilities:
>
> The first is we remove the filtering on time fields. There might not be much use cases for filtering on time since we specify the time range in the dashboard anyway, so its more than often part of the query and results are returned accordingly.
>
>  The second possibility is to convert the value to ISO format before filtering (for filter like Regex, or equal) or in other cases (for filter like greater), convert a filter expression to epoch time. This way the user could say Include Time Greater Than 2020-08-22, and they would get the desired result. To do that however I need to put the finger on the functions that are used to format the time values for display. If someone can point them out, I would investigate this second possibility.

There are also more subtleties to discuss if we want to implement filtering for the time field. Things like :

- In which format the user be able to input the time ?
- If the user inputs the time in ISO, should we keep the ISO format in the filter expression or should we convert to epoch so it becomes independent of the time zone ? Because otherwise it would filter differently in different time zones.
- When you enter a Regex, does it filter on the displayed value ? I guess the displayed, but then you are affected by the time zone again because your Regex will not be converted for each time zone.

So since I didn't get any comment on that and I didn't want to slow down the merge with additional discussion, I figured for now I would just remove the time option and keep that for a future PR. We could create a new issue or wait for a user to ask for its support, see if it's really a need.

Reply:
Got it! I agree - lets wait with this and open a separate PR for this if it is needed.